src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    public static final double ARRAY_GROWTH_VALUE = 1.5;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayIsFull();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        growIfArrayIsFull();
+        System.arraycopy(elementData, index, elementData, index + 1,
+                size - index);
+        elementData[index] = value;
+        size++;
+    }
+
+    private void checkIndexForAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add value to index " + index
+                    + ". Index is invalid.");
+        }
+    }
 
+    private void growIfArrayIsFull() {
+        if (size == elementData.length) {
+            Object[] grownElementData = new Object[(int) (elementData.length
+                                + (elementData.length * 1.5))];
```

Comment:
The logic for growing the array is incorrect. According to the task requirements, the array size should be increased by multiplying the current length by 1.5. The current implementation adds 1.5 times the length to itself, which is not correct. Please adjust this logic.